### DISCUSSION TITLE
Is there any interest in supporting Dark Mode in addition to the existing Light Mode?

### DISCUSSION THREAD

**Discussion by paul1956** [dotnet/winforms#4668] [Q&A] [answered]
This is not a proposal just a question as to what a plans might be to support Dark Mode and/or would there be interest is accepting any PR's that would support dark mode OR make it easier for others to support it. This is not Dark Mode instead of the existing Light Mode but the ability to switch back and forth at Runtime.

**Is your feature request related to a problem? Please describe.**

WinForms should have a native way to support Dark Mode or at least better support for it being added by 3rd parties. This might include exposing Border for example on ToolStrip controls. @dualbrain can add more specifics to this about specific issues if there is interest. 

**Describe the solution you'd like and alternatives you've considered**
    - More exposure of events and properties for controls contained in container controls
    - The simplest would be a Mode Property with Dark or Light and it would propagate to child controls or just a mode on each control.

**Will this feature affect UI controls?**    Yes

    If yes, describe the following:

        - Will VS Designer need to support the feature? Only to set initial Mode
        - What impact will it have on accessibility? None
        - Will this feature need to be localized or be localizable? No

**Comment by merriemcgaw** [comment, marked as answer]
There is definitely and interest on our end to fully support the native OS theming. At this point there is no public API that we can interact with in from WinForms. I'm working with the Windows team responsible for theming and we're hopeful that we'll be able to work together to create a solution for WinForms developers (not to mention native Win32 developers). There is a significant demand for this ability so that our WinForms apps have the native look/feel from Windows.

We're not ready to announce a release where this functionality might make an appearance. There are a lot of unknowns at this point, and it may take a while for us to sort it all out. It's also very significant undertaking that will require cross-company collaboration. Keep checking back for news.

**Reply by noseratio** [reply]
FWIW, I believe this is [how Chrome detects it](https://stackoverflow.com/a/51336913/1768303): 

> it looks like this option is not exposed to regular Win32 applications directly, however it can be set / retrieved through the `AppsUseLightTheme` key at the `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize` registry path.

Indeed, the amount of work would be hard to underestimate, but thanks for looking into it!

**Reply by RussKie** [reply]
We can only use public (i.e. documented) Win32 API. We're liaising with the Windows team to have required APIs available to us.
